spec/lib/response-handler.spec.js

```diff
@@ -0,0 +1,134 @@
+"use strict";
+
+describe("lib/response-handler", () => {
+    var bluebird, error, logger, parseLinkHeader, responseHandler, ShelfError, URI;
+
+    bluebird = require("bluebird");
+    error = require("../../lib/error")();
+    logger = require("../../lib/logger")();
+    parseLinkHeader = require("parse-link-header");
+    ShelfError = require("../../lib/shelf-error")();
+    URI = require("urijs");
+    responseHandler = require("../../lib/response-handler")(bluebird, error, logger, parseLinkHeader, ShelfError, URI);
+    describe(".isErrorCode()", () => {
+        it("returns false if the status code is within the right range", () => {
+            expect(responseHandler.isErrorCode(300)).toBe(false);
+        });
+        it("returns false if the status code is 200", () => {
+            expect(responseHandler.isErrorCode(200)).toBe(false);
+        });
+        it("returns false if the status code is 201", () => {
+            expect(responseHandler.isErrorCode(201)).toBe(false);
+        });
+        it("returns true if the status code is less than 200", () => {
+            expect(responseHandler.isErrorCode(150)).toBe(true);
+        });
+        it("returns true if the status code is greater than 399", () => {
+            expect(responseHandler.isErrorCode(404)).toBe(true);
+        });
+    });
+    describe(".createErrorForResponse()", () => {
+        var errorFromResponse, errorResponse;
+
+        it("returns any unknown errors", () => {
+            errorResponse = {
+                body: {
+                    code: "anUnknownError"
+                }
+            };
+            errorFromResponse = responseHandler.createErrorForResponse(errorResponse);
+            expect(errorFromResponse.message).toBe("UNKNOWN");
+            expect(errorFromResponse.code).toBe("UNKNOWN");
+        });
+        it("returns a timeout error", () => {
+            errorResponse = {
+                code: "exampleTimeoutError"
+            };
+            errorFromResponse = responseHandler.createErrorForResponse(errorResponse);
+            expect(errorFromResponse.message).toBe("Socket Timeout");
+            expect(errorFromResponse.code).toBe("ESOCKETTIMEDOUT");
+        });
+        it("returns known errors", () => {
```

Comment:
The intention of the test is to show that it returns errors other than `error.UNKNOWN`. That it's duplicate artifact error is only incidental.